### PR TITLE
Handle small reads

### PR DESCRIPTION
**Pull request by AudriusButkevicius**

### CODE REVIEW

**Comment by calmh** [comment]
Then normal behavior of a packet conn is that a given packet is only read once. That is, if I do a read into a buffer smaller than the actual packet I get the part of the packet that fits and the rest is simply discarded. So I think this can be made simpler, putting the rest of the packet back is incorrect and unnecessary. 

(This also indicates a bug somewhere else, that we are dropping data unintentionally by a short read?)

**Comment by AudriusButkevicius** [comment]
The short read in the panic comes from the quic library.

**Comment by AudriusButkevicius** [comment]
https://github.com/lucas-clemente/quic-go/blob/1e148c20c5cbb387e4019332ce3ee284d4304716/internal/protocol/protocol.go#L71 

Are kernel reads packet oriented as well? It' won't do something like join two packets into a larger continous packet?

**Comment by calmh** [comment]
Yes, they are, so that’s fine. I guess there may have been non-QUIC packets on the socket that were unexpected and nothing to worry about.

File: conn.go [hunk 1]
```diff
@@ -76,9 +76,13 @@ func (r *FilteredConn) ReadFrom(b []byte) (n int, addr net.Addr, err error) {
 	case <-timeout:
 		return 0, nil, errTimeout
 	case pkt := <-r.recvBuffer:
-		copy(b[:pkt.n], pkt.buf)
+		n := pkt.n
+		if cap(b) < n {
```
Comment (calmh) [inline conn.go]: I don’t think you can copy past the slice len, nor are expected to, so s/cap/len/?

**Comment by calmh** [comment]
BTW re. the link to the comment about 1472 bytes max packet size above that’s only relevant if the same constant is used on the sending side and. UDP packets themselves are not limited to MTU and can be up to slightly less than 64 KiB. I’m guessing this implementation won’t send such packets though (I only saw 1200 bytes on in pcap early on)

File: conn.go (hunk 1 above)
Comment (AudriusButkevicius) [reply conn.go]: https://play.golang.org/p/nuyKA9ztEUe

**Comment by AudriusButkevicius** [comment]
The assumption is that they do respect that, so perhaps just some garbage arriving?

**Comment by calmh** [comment]
Yes I think so as well.

File: conn.go (hunk 1 above)
Comment (AudriusButkevicius) [reply conn.go]: right, that won't work, as it will have no effect on the slice the caller has.

**Comment by AudriusButkevicius** [comment]
Looks good?

**Pull request merged**